resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/repository/ResourceRepository.java

```diff
@@ -18,40 +18,41 @@
 import org.springframework.data.repository.CrudRepository;
 
 import uk.gov.gchq.palisade.service.resource.domain.ResourceEntity;
+import uk.gov.gchq.palisade.service.resource.service.FunctionalIterator;
 
-import java.util.Optional;
-import java.util.stream.Stream;
-import java.util.stream.StreamSupport;
+import java.util.Iterator;
 
 /**
  * Low-level requirement for a database used for persistence, see {@link ResourceEntity}
  * for more details
+ *
+ * @implNote In the future consider changing this to a {@link org.springframework.data.repository.reactive.ReactiveCrudRepository}
```

Comment:
Does a techDebt ticket need to be created for this?

Reply:
Ticket to be created in the Backlog for a more in-depth investigation.